### ISSUE TITLE
Class-based DTOs are not yet supported.

### ISSUE THREAD

**Issue by anarchical** [spring-projects/spring-data-jpa#3702]
springboot version: v3.3.7-SNAPSHOT

```
Page<SchoolInterface> pageResult = schoolRepository.findBy(specification, q -> q.as(SchoolInterface.class));
```
When projection target is Interface, it works well.

But error occurred when I change the projection target to DTOs(normal class or record type)

java.lang.UnsupportedOperationException: Class-based DTOs are not yet supported.

I hope to know why must use interface.

**Comment by christophstrobl** [comment]
likely to be closed by #3654

**Comment by mp911de** [comment]
Duplicate of #2327
